File: app/src/main/java/com/jobtick/activities/PaymentHistoryActivity.java
package com.jobtick.activities;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.RadioButton;
import android.widget.RadioGroup;

import androidx.annotation.Nullable;
import androidx.viewpager.widget.ViewPager;

import com.google.android.material.appbar.MaterialToolbar;
import com.jobtick.R;
import android.annotation.SuppressLint;

import com.jobtick.adapers.SectionsPagerAdapter;
import com.jobtick.fragments.PaymentPosterFragment;
import com.jobtick.fragments.PaymentTickerFragment;

import butterknife.BindView;
import butterknife.ButterKnife;

public class PaymentHistoryActivity extends ActivityBase {

    @SuppressLint("NonConstantResourceId")
    @BindView(R.id.toolbar)
    MaterialToolbar toolbar;

    @SuppressLint("NonConstantResourceId")
    @BindView(R.id.view_pager)
    ViewPager viewPager;
    @SuppressLint("NonConstantResourceId")
    @BindView(R.id.rb_outgoing)
    RadioButton rbOutgoing;
    @SuppressLint("NonConstantResourceId")
    @BindView(R.id.rb_earned)
    RadioButton rbEarned;
    @SuppressLint("NonConstantResourceId")
    @BindView(R.id.rg_outgoing_earned)
    RadioGroup rgOutgoingEarned;

    ViewPager.OnPageChangeListener viewPagerPageChangeListener = new ViewPager.OnPageChangeListener() {
        @Override
        public void onPageSelected(final int position) {
            switch (position) {
                case 0:
                    rbEarned.setChecked(false);
                    rbOutgoing.setChecked(true);
                    break;
                case 1:
                    rbEarned.setChecked(true);
                    rbOutgoing.setChecked(false);
                    break;
                case 2:

                    break;
            }
        }

        @Override
        public void onPageScrolled(int arg0, float arg1, int arg2) {
        }

        @Override
        public void onPageScrollStateChanged(int arg0) {
        }
    };

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_payment_history);
        ButterKnife.bind(this);
        initComponent();
        initToolbar();
    }

    private void initComponent() {
        setupViewPager(viewPager);
        viewPager.addOnPageChangeListener(viewPagerPageChangeListener);
        viewPager.setCurrentItem(0);
        viewPager.setOffscreenPageLimit(2);
        clickEvent();
    }

    private void initToolbar() {
        toolbar.setTitle("Payment History");
        toolbar.setNavigationIcon(R.drawable.ic_back);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_payment_history, menu);
        return true;
    }


    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                onBackPressed();
                break;
            case R.id.nav_setting:
                startActivity(new Intent(this, PaymentSettingsActivity.class));
                break;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
    }


    private void clickEvent() {
        rgOutgoingEarned.setOnCheckedChangeListener((group, checkedId) -> {
            switch (checkedId) {
                case R.id.rb_outgoing:
                    viewPager.setCurrentItem(0);
                    break;

                case R.id.rb_earned:
                    viewPager.setCurrentItem(1);
                    break;
            }
        });
    }

    private void setupViewPager(ViewPager viewPager) {
        SectionsPagerAdapter adapter = new SectionsPagerAdapter(getSupportFragmentManager());
        adapter.addFragment(PaymentPosterFragment.newInstance(), getResources().getString(R.string.date_time));
        adapter.addFragment(PaymentTickerFragment.newInstance(), getResources().getString(R.string.details));
        viewPager.setAdapter(adapter);
    }
}

File: app/src/main/java/com/jobtick/android/activities/PaymentHistoryActivity.kt
package com.jobtick.android.activities

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.RadioButton
import android.widget.RadioGroup
import androidx.viewpager.widget.ViewPager
import androidx.viewpager.widget.ViewPager.OnPageChangeListener
import com.google.android.material.appbar.MaterialToolbar
import com.jobtick.android.R
import com.jobtick.android.adapers.SectionsPagerAdapter
import com.jobtick.android.fragments.PaymentPosterFragment
import com.jobtick.android.fragments.PaymentTickerFragment

class PaymentHistoryActivity : ActivityBase() {
    private lateinit var toolbar: MaterialToolbar
    private lateinit var viewPager: ViewPager
    private lateinit var rbOutgoing: RadioButton
    private lateinit var rbEarned: RadioButton
    private lateinit var rgOutgoingEarned: RadioGroup
    private var viewPagerPageChangeListener: OnPageChangeListener = object : OnPageChangeListener {
        override fun onPageSelected(position: Int) {
            when (position) {
                0 -> {
                    rbEarned.isChecked = false
                    rbOutgoing.isChecked = true
                }
                1 -> {
                    rbEarned.isChecked = true
                    rbOutgoing.isChecked = false
                }
                2 -> {
                }
            }
        }

        override fun onPageScrolled(arg0: Int, arg1: Float, arg2: Int) {}
        override fun onPageScrollStateChanged(arg0: Int) {}
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_payment_history)
        setIDs()
        initComponent()
        initToolbar()
    }

    private fun setIDs() {
        toolbar = findViewById(R.id.toolbar)
        viewPager = findViewById(R.id.view_pager)
        rbOutgoing = findViewById(R.id.rb_outgoing)
        rbEarned = findViewById(R.id.rb_earned)
        rgOutgoingEarned = findViewById(R.id.rg_outgoing_earned)
    }

    private fun initComponent() {
        setupViewPager(viewPager)
        viewPager.addOnPageChangeListener(viewPagerPageChangeListener)
        viewPager.currentItem = 0
        viewPager.offscreenPageLimit = 2
        clickEvent()
    }

    private fun initToolbar() {
        toolbar.title = "Payment History"
        toolbar.setNavigationIcon(R.drawable.ic_back)
        setSupportActionBar(toolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_payment_history, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> onBackPressed()
            R.id.nav_setting -> startActivity(Intent(this, PaymentSettingsActivity::class.java))
        }
        return super.onOptionsItemSelected(item)
    }

    private fun clickEvent() {
        rgOutgoingEarned.setOnCheckedChangeListener { group: RadioGroup?, checkedId: Int ->
            when (checkedId) {
                R.id.rb_outgoing -> viewPager.currentItem = 0
                R.id.rb_earned -> viewPager.currentItem = 1
            }
        }
    }

    private fun setupViewPager(viewPager: ViewPager?) {
        val adapter = SectionsPagerAdapter(supportFragmentManager)
        adapter.addFragment(PaymentPosterFragment.newInstance(), resources.getString(R.string.date_time))
        adapter.addFragment(PaymentTickerFragment.newInstance(), resources.getString(R.string.details))
        viewPager!!.adapter = adapter
    }
}